guide/token-setup-and-validation.md

```diff
@@ -0,0 +1,351 @@
+            //This is required if using introspection on resource servers
+            //This is not needed if resource servers will use shared ASP.NET Core DataProtection
+            //options.EnableIntrospectionEndpoint("/connect/introspect");
+        })
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+## Default token validation
+
+### Authorization server
+```c#
+// Startup.cs
+
+//ConfigureServices
+services.AddOpenIddict()
+        .AddCore(...)
+
+        .AddServer(...)
+
+        .AddValidation();
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Resource server shared ASP.NET Core DataProtection
+```c#
+// Startup.cs
+
+//ConfigureServices
+//If tokens need to be validated in a separate resource server, configure a shared ASP.NET Core DataProtection with shared key store and application name
+//See https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-2.1&tabs=aspnetcore2x#setapplicationname
+services.AddDataProtection()
+    .PersistKeysToFileSystem(new System.IO.DirectoryInfo(@"[UNC PATH]"))
+    .SetApplicationName("[APP NAME]");
+
+services.AddOpenIddict()
+        //This adds a "Bearer" authentication scheme
+        .AddValidation();
+
+//Optionally set Bearer token authentication as default
+//services.AddAuthentication("Bearer");
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+### Resource server - introspection
+```c#
+// Introspection requires a request to auth server for every token so shared ASP.NET Core DataProtection is preferred
+// Startup.cs
+
+//ConfigureServices
+services.AddAuthentication(options =>
+        {
+            options.DefaultScheme = OAuthIntrospectionDefaults.AuthenticationScheme;
+        })
+
+        .AddOAuthIntrospection(options =>
+        {
+            //example settings
+            options.Authority = new Uri("http://localhost:12345/");
+            options.Audiences.Add("resource-server-1");
+            options.ClientId = "resource-server-1";
+            options.ClientSecret = "846B62D0-DEF9-4215-A99D-86E6B8DAB342";
+            options.RequireHttpsMetadata = false;
+
+            // Note: you can override the default name and role claims:
+            // options.NameClaimType = "custom_name_claim";
+            // options.RoleClaimType = "custom_role_claim";
+        });
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Api controller
+```c#
+// Note: both OpenIddictValidationDefaults.AuthenticationScheme and JwtBearerDefaults.AuthenticationScheme are "Bearer"
+//If "Bearer" is not the default authentication scheme then specify it here.
+//If you get a 302 redirect to login page instead of a 401 Unauthorized then Cookie authentication is handling the request
+//so "Bearer" must be specified
+[Authorize(AuthenticationSchemes = "Bearer")]
```

Comment:
Since this section is dedicated to introspection, I'd probably go with `OAuthIntrospectionDefaults.AuthenticationScheme` and remove the associated comment.